agent/csi/mod_csi_docker.go

```diff
@@ -790,24 +822,41 @@ func (d *dockerBridge) Mount(
 	// Create a new CSI Node client.
 	nc := csi.NewNodeClient(c)
 
-	// Next, public the volume at the Controller level. To do that this
-	// Node's ID is required.
-	nodeID, err := gocsi.GetNodeID(d.ctx, nc, csiVersion)
+	// Check if the CSI plugin supports ControllerPublishVolume
+	caps, err := gocsi.ControllerGetCapabilities(d.ctx, cc, csiVersion)
 	if err != nil {
 		return nil, err
 	}
 
+	var (
+		volCap  *csi.VolumeCapability
+		pubInfo *csi.PublishVolumeInfo
+	)
+
 	// Create a new volume capability for publishing the volume
 	// via the Controller and Node.
-	volCap := newVolumeCapability(d.fsType)
+	if d.isCSINFS() {
+		volCap = newVolumeCapability("nfs")
```

Comment:
While this isn't explicitly necessary, it was weird to see a request coming in with a fstype of "ext4". If the CSI-NFS plugin saw this via `VolumeValidateCapabilities` it would reject it.

Reply:
Good catch.